pandas/_libs/hashtable_func_helper.pxi.in

```diff
@@ -43,7 +43,13 @@ cdef value_count_{{dtype}}(const {{dtype}}_t[:] values, bint dropna, const uint8
         # Don't use Py_ssize_t, since table.n_buckets is unsigned
         khiter_t k
 
-        {{c_type}} val
+        # without explicit initialization the compiler detects that val could
```

Comment:
This is extremely wonky, but the compiler is right that `val` would be uninitialized if `n==0`. I think it would take a decent refactor to make Cython generate something else that the compiler would be comfortable with